src/task/extra.rs

```diff
@@ -0,0 +1,129 @@
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+#[cfg_attr(feature = "serde-support", derive(Serialize, Deserialize))]
+pub struct Extra {
+    #[cfg_attr(feature = "serde-support", serde(flatten))]
+    pub inner: super::Task,
+    #[cfg_attr(feature = "serde-support", serde(default))]
+    pub note: super::Note,
+    pub recurrence: Option<super::Recurrence>,
+    #[cfg_attr(feature = "serde-support", serde(default))]
+    pub flagged: bool,
+}
+
+impl Extra {
+    fn note(task: &super::Task) -> super::Note {
+        let tag = match ::std::env::var("TODO_NOTE_TAG") {
+            Ok(tag) => tag,
+            Err(_) => "note".to_owned(),
+        };
+
+        if let Some(file) = task.tags.get(&tag) {
+            super::Note::from_file(file)
+        } else {
+            super::Note::None
+        }
+    }
+
+    pub fn complete(&mut self) {
+        let today = ::chrono::Local::now().date().naive_local();
+
+        self.finished = true;
+        self.finish_date = Some(today);
+    }
+
+    pub fn uncomplete(&mut self) {
+        self.finished = false;
+        self.finish_date = None;
+    }
+}
+
+impl ::std::str::FromStr for Extra {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, ()> {
```

Comment:
Looks like you could extract a `Extra::from_task(Task)` function from this, since it could come useful in other contexts.